src/components/datepicker/DatePickerInput.tsx

```diff
@@ -123,22 +123,19 @@ const DatePickerInput: GenericComponent<DatePickerInputProps> = ({
   };
 
   const handleInputChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-    const value = event.target.value;
-    const date = parseMultiFormatsDate(value, ALLOWED_DATE_FORMATS, locale);
     setDisplayError(false);
-    if (date) {
-      setSelectedDate(date);
-      setInputValue(getFormattedDateString(date));
-    } else {
-      setInputValue(value);
-    }
+    const value = event.target.value;
```

Comment:
Was thinking that it would be nice to keep doing `setSelectedDate(date)` in here when it's a valid date, so that you see the selected date changing in the picker while typing, and you don't need to blur it. 

```ts
  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    setDisplayError(false);
    const value = event.target.value;
    setInputValue(value);

    const date = parseMultiFormatsDate(value, ALLOWED_DATE_FORMATS, locale);

    if (date) {
      setSelectedDate(date);
    }
  };
```

But this is not working at the moment, because you removed this part in DatePicker. 
```ts
  useEffect(() => {
    setSelectedDate(others.selectedDate);
    setSelectedMonth(others.selectedDate);
  }, [others.selectedDate]);
```

What's the reason that you removed this?

Reply:
Because setting date in `inputChange` was causing that bug you recorded on video, why it was happening is written [here](https://github.com/teamleadercrm/ui/pull/2430#issuecomment-1305492973). We want to support all allowed formats (in description of PR) + also legacy datepicker input  works similar, so those are the reason we chose this approach, together with design. Of course best way would be to support changing of date on `inputChange`, but then user has that bug, which is a no go :/

Reply:
The exact reason that it was happening is not because you were setting the date, but because you were setting the inputValue. More specifically, you were setting it to something else than what the input actually had, which causes the cursor to be moved. For example, if the input value was `2/11/2022`, you parsed the date and then set the inputValue to `02/11/2022`. 

If you don't do that parsing on inputChange, it actually works fine (I have it working with the code from my previous comment). We need to be careful with executing the `onChange` handler though, I didn't look into that yet.


https://user-images.githubusercontent.com/330765/200489954-69754e78-52d8-4d7c-a6a3-e47e4805f67c.mp4

Reply:
Sorry I have expressed it incorrectly in previous comment. It is indeed because of setting inputValue, however that is desired behaviour, since we accept many formats but once date is set/picked, we want to displat it only in default format. I will try to go one more round with design, maybe we find some compromise there. Thanks